### ISSUE TITLE
Support piecewise function definition

### ISSUE THREAD

**Issue by Ailrun** [facebookincubator/retrie#7]
Current retrie picks the first piece of a piecewise function definition when trying to unfold the funciton. For example, with this code
```haskell
module Test where

piece :: Bool -> Int
piece True = 1
piece False = 0

usePiece :: Bool -> Int
usePiece b = piece b
```
if I execute `retrie -u "Test.piece" --target-file Test.hs`, it returns
```haskell
module Test where

piece :: Bool -> Int
piece True = 1
piece False = 0

usePiece :: Bool -> Int
usePiece b = (\ True -> 1) b
```
which is not semantically same.

**Comment by xich** [comment]
So we used to have something to handle this. From the single definition, it would generate three rewrites:

```
  piece True = 1

  piece False = 0

  forall x. piece x = case x of
    True -> 1
    False -> 0
```

I removed it at some point because the function-to-case conversion was buggy and lead to some strange stuff. Things are more mature now, so I could try adding it back.

Does that seem like the behavior you would expect?

**Comment by Ailrun** [comment]
Yes, handling the example using `case` sounds reasonable for me.
